### PR TITLE
Reviewed `LogisticRegression`

### PR DESCRIPTION
**Pull request by Ce11an**
## What does this PR do?

Reviewed `LogisticRegression` as part of #839

- Created protective methods to share code.
- Using `nn.CrossEntropyLoss`.
- Updated docstrings.
- Fixed previously stale issues #793 and #794.
- Using `self.log`: #405.

Still need to do the following before merging:
- [x] Update `tests`
- [ ] Discuss the coupling between the Model and the `accuracy` metric.
- [ ] Discuss the coupling between the Model and the loss function.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [x] Did you verify new and **existing tests** pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Of course 🥳

### CODE REVIEW

**Comment by Ce11an** [comment]
Hey all 👋🏻

If I could have an initial review that would be great 😄 

Are we happy to keep `accuracy` within the class? Ideally, I don't want to couple it, so if you have any suggestions/examples of Lighting code that solves this issue?

Also, do you have any recommendations on how to properly unit-test sub-classes of `LightningModule`? 

Thanks!!

File: src/pl_bolts/models/regression/logistic_regression.py [hunk 1]
```diff
@@ -30,85 +29,157 @@ def __init__(
     ) -> None:
         """
```
Comment (Borda) [inline src/pl_bolts/models/regression/logistic_regression.py:30]:
```suggestion
        """Logistic Regression

```

File: src/pl_bolts/models/regression/logistic_regression.py [hunk 2]
```diff
@@ -30,85 +29,157 @@ def __init__(
...
+            criterion: Cross-Entropy loss function.
+            optimizer: Model optimizer to use.
         """
         super().__init__()
         self.save_hyperparameters()
         self.optimizer = optimizer
-
-        self.linear = nn.Linear(in_features=self.hparams.input_dim, out_features=self.hparams.num_classes, bias=bias)
+        self.criterion: nn.CrossEntropyLoss = nn.CrossEntropyLoss()
+        self.linear: nn.Linear = nn.Linear(
```
Comment (Borda) [inline src/pl_bolts/models/regression/logistic_regression.py:49]:
```suggestion
        self.criterion = nn.CrossEntropyLoss()
        self.linear = nn.Linear(
```

**Review by Borda — APPROVED** [review]

File: src/pl_bolts/models/regression/logistic_regression.py [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
-from torchmetrics.functional import accuracy
+from torchmetrics import functional
 
 from pl_bolts.utils.stability import under_review
 
 
-@under_review()
 class LogisticRegression(LightningModule):
-    """Logistic regression model."""
+    """Logistic Regression Model."""
```
Comment (Borda) [inline src/pl_bolts/models/regression/logistic_regression.py:17]:
```suggestion
    """Logistic Regression Model."""
    
    criterion: nn.CrossEntropyLoss
    linear: nn.Linear
```
